File: core/src/main/java/pcb/uwu/unit/derived/optics/LuminanceUnit.java
package pcb.uwu.unit.derived.optics;

import pcb.uwu.core.CompositeUnit;
import pcb.uwu.core.UnitCounter;
import pcb.uwu.unit.base.LengthUnit;
import pcb.uwu.unit.base.LuminousIntensityUnit;

public class LuminanceUnit extends CompositeUnit {

	// region base constructor

	public LuminanceUnit(LuminousIntensityUnit luminousIntensityUnit, LengthUnit lengthUnit) {
		super(new UnitCounter()
				.major(luminousIntensityUnit.getUnitCounter())
				.minor(lengthUnit, 2));
	}

	// endregion

}

File: core/src/main/java/pcb/uwu/unit/derived/optics/LuminanceUnit.kt
package pcb.uwu.unit.derived.optics

import pcb.uwu.core.CompositeUnit
import pcb.uwu.core.UnitCounter
import pcb.uwu.unit.base.LuminousIntensityUnit
import pcb.uwu.unit.derived.fundamental.AreaUnit

open class LuminanceUnit
    : CompositeUnit
{
    @Suppress("ConvertSecondaryConstructorToPrimary")
    constructor(luminousIntensityUnit: LuminousIntensityUnit,
                areaUnit: AreaUnit)
            : super(UnitCounter()
                        .major(luminousIntensityUnit)
                        .minor(areaUnit))
}